### PR TITLE
Cancel active and pending proposals on runtime upgrade.

### PR DESCRIPTION
**Pull request by shamil-gadelshin**
[Main issue](https://github.com/Joystream/joystream/issues/2008)

### CODE REVIEW

File: runtime-modules/proposals/engine/src/tests/mod.rs [hunk 1]
```diff
@@ -901,6 +901,126 @@ fn proposal_execution_postponed_because_of_grace_period() {
...
+
+        // A proposal for pending execution check.
+        let dummy_proposal =
+            DummyProposalFixture::default().with_parameters(parameters_fixture.params());
+        let proposal_id = dummy_proposal.create_proposal_and_assert(Ok(1)).unwrap();
+
+        // A proposal for active status check.
+        let dummy_proposal =
+            DummyProposalFixture::default().with_parameters(parameters_fixture.params());
+        let proposal_id2 = dummy_proposal.create_proposal_and_assert(Ok(2)).unwrap();
```
Comment (conectado) [inline runtime-modules/proposals/engine/src/tests/mod.rs]: We could use more descriptive names for these proposals ids such as `proposal_id_pending_execution` and `proposal_id_active`.

File: runtime-modules/proposals/engine/src/tests/mod.rs [hunk 2]
```diff
@@ -901,6 +901,126 @@ fn proposal_execution_postponed_because_of_grace_period() {
...
+        vote_generator.vote_and_assert_ok(VoteKind::Approve);
+        vote_generator.vote_and_assert_ok(VoteKind::Approve);
+        vote_generator.vote_and_assert_ok(VoteKind::Approve);
+
+        run_to_block(3);
+
+        let proposal = <crate::Proposals<Test>>::get(proposal_id);
+
+        // ensure proposal has pending execution status
+        assert_eq!(
```
Comment (conectado) [inline runtime-modules/proposals/engine/src/tests/mod.rs:937]: Shouldn't we do the same check for the other proposal?

File: runtime-modules/proposals/engine/src/tests/mod.rs [hunk 3]
```diff
@@ -901,6 +901,126 @@ fn proposal_execution_postponed_because_of_grace_period() {
...
+        assert_eq!(<ActiveProposalCount>::get(), 1);
+
+        let mut vote_generator = VoteGenerator::new(proposal_id);
+        vote_generator.vote_and_assert_ok(VoteKind::Approve);
+        vote_generator.vote_and_assert_ok(VoteKind::Approve);
+        vote_generator.vote_and_assert_ok(VoteKind::Approve);
+        vote_generator.vote_and_assert_ok(VoteKind::Approve);
+
+        run_to_block_and_finalize(2);
+
```
Comment (conectado) [inline runtime-modules/proposals/engine/src/tests/mod.rs:1031]: Can we add a check here to be sure that the proposal is pending constitutionality?

**Review by conectado — CHANGES_REQUESTED** [review]
Look good. Just a couple of minor comments also please fix the compilation with `cargo test --all --feature runtime-benchmarks`

File: runtime-modules/proposals/engine/src/tests/mod.rs (hunk 1 above)
Comment (shamil-gadelshin) [reply runtime-modules/proposals/engine/src/tests/mod.rs]: Sure.

File: runtime-modules/proposals/engine/src/tests/mod.rs (hunk 2 above)
Comment (shamil-gadelshin) [reply runtime-modules/proposals/engine/src/tests/mod.rs:937]: Sure.

File: runtime-modules/proposals/engine/src/tests/mod.rs (hunk 3 above)
Comment (shamil-gadelshin) [reply runtime-modules/proposals/engine/src/tests/mod.rs:1031]: Sure.

**Review by conectado — APPROVED** [review]
LGTM

**Pull request merged**